TacTok/models/bpe.py

```diff
@@ -0,0 +1,118 @@
+
+from collections import Counter
+from typing import List, Dict, Tuple, Optional
+import typing
+
+
```

Comment:
Can you include a few comments explaining this, and link to the BPE tutorial we sourced the algorithm from?